cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaReconciler.java

```diff
@@ -283,6 +293,66 @@ public Future<Void> reconcile(KafkaStatus kafkaStatus, Clock clock)    {
                 .compose(i -> updateKafkaMetadataState(kafkaStatus));
     }
 
+    private Future<Void> updateKafkaAutoRebalanceStatus(KafkaStatus kafkaStatus) {
+        // we are going to save a consistent list of added notes (between the ones already in the status and the newly added ones)
+        // into the KafkaStatus autorebalance field to be passed to the auto-rebalancing reconciler.
+        // It's needed in case nodes are scaled up but reconciliation fails and on the next one the operator has to know where to rebalance
+
+        Set<Integer> newAddedBrokers = new LinkedHashSet<>(kafka.addedBrokerNodes());
+        if (kafkaAutoRebalanceStatus == null) {
+            // no already stored added nodes, adding only the new ones from current reconciliation
+            if (!newAddedBrokers.isEmpty()) {
+                kafkaStatus.setAutoRebalance(
+                        new KafkaAutoRebalanceStatusBuilder()
+                                .withModes(
+                                        new KafkaAutoRebalanceModeBrokersBuilder()
+                                                .withMode(KafkaRebalanceMode.ADD_BROKERS)
+                                                .withBrokers(kafka.addedBrokerNodes().stream().toList())
+                                                .build()
+                                )
+                                .build());
+            }
+            return Future.succeededFuture();
+        }
```

Comment:
You should use `else` to have properly checked returns in the future.

Reply:
Sorry what do you mean by `properly checked returns in the future` ? I think we use this logic about exiting a method (and returning a Future this way) even in other places instead of having if ... else ... if ... else which increase the cyclomatic complexity  (and then most of the time we use a suppress warning for it).

Reply:
When you structure the code like this:

```
if {
  ...
  return ...;
}

if {
  ...
}

if {
  ...
  return ...;
}

return ...;
```

You lose any changes that every path of the code properly returns as desired. If you instead structure things more like this:

```
if {
  ...
  return ...;
} else {
  if {
    ...
  }
  
  if {
    ...
    return ...;
  } else {
    return ...;
  }
}
```

You have much better support from IDE / compiler to check that you properly return. The code might work fine today, but next week someone might add new `if`, forget to add return and the code will on paper still work.

Reply:
Tbh my initial code was structured as you showed above but I found it less readable than the current one. Of course, the point around the IDE check is useful. I will revert to the previous shape.

Reply:
I refactored the code as requested.